rust-src/aggregate_sig/src/aggregate_sig.rs

```diff
@@ -171,6 +171,55 @@ pub fn verify_aggregate_sig<P: Pairing>(
     P::pair(&signature.0, &P::G2::one_point()) == product
 }
 
+/// Verifies an aggregate signature on pairs `(messages m_i, set_i)` `for
+/// i=1..n` but where `set_i` denotes the set of public keys corresponding to
+/// the secret keys that signed m_i. This implements a combination of
+/// AggregateVerify from Section 3.1.1 and FastAggregateVerify from Section
+/// 3.3.4 of https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05#section-3.1.1.
+///
+/// In particular, this function returns:
+///
+/// pairing(sig, g_2) == ∏ᵢ pairing(H(mᵢ), ∑ⱼ pkᵢⱼ)
+///
+/// Each message should be associated with at least one key. However, the result
+/// of having a message with no keys will be the same as if the message was not
+/// included.
+///
+/// For security, the holder of a key must be required to prove knowledge of the
+/// secret key.
```

Comment:
This is nitpicky, but that's not really necessary. It is sufficient, but as long as people don't cheat and reuse other's keys they don't have to prove it either.

Reply:
I think the formulation is correct. If people don't cheat, just sending the plain message without any signature would be "secure", so you cannot assume people don't cheat when talking about security.